lib/validator.js

```diff
@@ -15,48 +54,131 @@ var inMapping = {
+            if (param.required) {
+                reqPartSchema.required = reqPartSchema.required || [];
+                reqPartSchema.required.push(param.name);
+            }
+        } else {
+            if (param.schema) {
+                schema.properties.body = param.schema;
+            } else {
+                schema.properties.body = {
+                    type: 'object'
+                };
+            }
+            if (param.required) {
+                schema.required = schema.required || [];
+                schema.required.push('body');
+            }
         }
     });
 
     return schema;
-}
+};
 
-var Validator = function(parameters) {
-    this.parameters = parameters;
-    this._validatorFunc = ajv.compile(convertToJsonSchema(parameters));
+/**
+ * Creates a function, that tries to coerce types of parameters in the
+ * incoming request, according to the provided parameters specification.
+ *
+ * @param parameters {Array} parameters swagger specification
+ * @returns {Function(req, HTTPError)} cersion function
+ * @private
+ */
+Validator.prototype._createTypeCoercionFunc = function(parameters) {
+    var code = '';
+    parameters.forEach(function(param) {
+        if (param.type === 'string') {
+            // Don't need to process strings
+            return;
+        }
+        var paramAccessor = 'req.' + inMapping[param.in] + '["' + param.name + '"]';
+        var paramCoercionCode;
+        var errorNotifier = 'throw new HTTPError({status:400,body:{type:"bad_request",'
+                + 'title:"data.' + inMapping[param.in] + '.' + param.name
+                + ' should be ' + param.type + '"}});\n';
+
+        switch (param.type) {
+            case 'integer':
+                paramCoercionCode = paramAccessor + ' = parseInt(' + paramAccessor + ');\n'
+                    + 'if (!Number.isInteger(' + paramAccessor + ')) {\n' + errorNotifier + '}\n';
+                break;
+            case 'number':
+                paramCoercionCode = paramAccessor + ' = parseFloat(' + paramAccessor + ');\n'
+                    + 'if (Number.isNaN(' + paramAccessor + ')) {\n' + errorNotifier + '}\n';
+                break;
+            case 'boolean':
+                paramCoercionCode = 'if(!/^(?:true)|(?:false)|(?:1)|(?:0)$/.test('
+                    + paramAccessor + '.toString())) {\n'
+                    + errorNotifier + '}\n'
+                    + paramAccessor + ' = /^(?:true)|(?:1)$/.test('
+                    + paramAccessor + '.toString());\n';
+        }
+
+        if (!paramCoercionCode) {
+            return;
+        }
+
+        if (!param.required) {
+            // If parameter is not required, don't try to coerce "undefined"
+            paramCoercionCode = 'if (' + paramAccessor + ' !== undefined) {\n'
+                + paramCoercionCode + '}\n';
+        }
+
+        code += paramCoercionCode;
+    });
+    if (code && code.trim()) {
+        code += 'return req;\n';
```

Comment:
Nit: add a space or a new line before `return` since you're trimming it in the line above.